File: re-ssim-service/src/main/java/com/shinonometn/re/ssim/service/caterpillar/repository/ImportTaskRepository.java
package com.shinonometn.re.ssim.service.caterpillar.repository;

import com.shinonometn.re.ssim.service.caterpillar.entity.ImportTask;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface ImportTaskRepository extends MongoRepository<ImportTask,String>{
}

File: re-ssim-services/re-ssim-data-kingo-application/src/main/java/com/shinonometn/re/ssim/data/kingo/application/repository/ImportTaskRepository.kt
package com.shinonometn.re.ssim.data.kingo.application.repository

import com.shinonometn.re.ssim.data.kingo.application.entity.ImportTask
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ImportTaskRepository : JpaRepository<ImportTask, Int> {

    @Query("select it from ImportTask it where it.captureTaskId = ?1")
    fun findByCaptureTaskId(id: Int): Optional<ImportTask>

    @Modifying
    fun deleteByCaptureTaskId(id: Int)
}
